tools/autograd/gen_variable_type.py

```diff
@@ -741,8 +832,10 @@ def requires_unpack(arg):
             body.append(UNPACK_OPTIONS.substitute(arg_name=arg['name']))
 
         unpacked_args.append(arg['name'] + '_')
+        unpacked_args_dynamic_type[arg['name'] + '_'] = dynamic_type
 
     env['unpacked_args'] = unpacked_args
+    env['unpacked_args_dynamic_type'] = unpacked_args_dynamic_type
```

Comment:
do you need to write this?  You can't look this up later by name?

Reply:
There seems to be no O(1) lookup table from unpacked_arg name to simple_type, and we likely have to build one.